src/util-thash.c

```diff
@@ -291,10 +291,13 @@ static void THashInitConfig(THashTableContext *ctx, const char *cnf_prefix)
 }
 
 THashTableContext* THashInit(const char *cnf_prefix, size_t data_size,
-    int (*DataSet)(void *, void *),
+     int (*DataSet)(void *, void *),
      void (*DataFree)(void *),
      uint32_t (*DataHash)(void *),
-     bool (*DataCompare)(void *, void *))
+     bool (*DataCompare)(void *, void *),
+     bool reset_memcap,
```

Comment:
doesn't look like its used, or needed either

Reply:
This is for the special case of `load`, since we want to make sure that everything from the dataset file is loaded.
if memcap is defined in rule/yaml, set that
else check if load was the cmd (with this param) and let the memcap be `UINT64_MAX`
if it wasn't a load cmd, set memcap to `THASH_DEFAULT_MEMCAP`.

Should this not be done?

Reply:
ah yes I missed the actual use, please ignore my comment